### PR TITLE
--git-ref: Fail fast when ref does not resolve

### PR DESCRIPTION
**Pull request by petr-muller**
Previously, `ci-operator` attempted to continue even when a ref passed
did not resolve to an existing commit, leading to later builds failing,
because cloneref did not fetch any code. This change makes `ci-operator`
to fail fast in this situation.

Resolves: openshift/release#1096

### CODE REVIEW

**Comment by petr-muller** [comment]
/assign @stevekuznetsov @smarterclayton 

@smarterclayton, you authored the error recovery attempt there, any thoughts?

File: cmd/ci-operator/main.go [hunk 1]
```diff
@@ -752,7 +752,7 @@ func jobSpecFromGitRef(ref string) (*steps.JobSpec, error) {
 	out, err := exec.Command("git", "ls-remote", fmt.Sprintf("https://github.com/%s/%s.git", prefix[0], prefix[1]), parts[1]).Output()
 	sha := strings.Split(strings.Split(string(out), "\n")[0], "\t")[0]
 	if len(sha) == 0 || err != nil {
-		return &steps.JobSpec{Type: steps.PeriodicJob, Job: "dev", Refs: steps.Refs{Org: prefix[0], Repo: prefix[1], BaseSHA: parts[1]}}, nil
+		return nil, fmt.Errorf("ref '%s' does not point to any commit in '%s'", parts[1], parts[0])
```
Comment (stevekuznetsov) [inline cmd/ci-operator/main.go]: Can we surface the `err` here?

Comment (petr-muller) [reply cmd/ci-operator/main.go]: Good idea.

**Comment by petr-muller** [comment]
Wait a bit, this is slightly tricky - there are different error conditions ($JOB_SPEC not being set, problematic `--git-ref` arguments) and their combination, and I do not want to show misleading error messages. I'll finish this tomorrow.

**Comment by petr-muller** [comment]
Ready for re-review

**Pull request merged**